packages/belos/tpetra/example/GCRODR/GCRODRTpetraExFile.cpp

```diff
@@ -0,0 +1,252 @@
+    }
+    if (debug) {
+      verbosity += Belos::Debug;
+    }
+    belosList.set( "Verbosity", verbosity );
+
+    // Construct an unpreconditioned linear problem instance.
+    Belos::LinearProblem<ST,MV,OP> problem( A, X, B );
+    bool set = problem.setProblem();
+    if (set == false) {
+      if (proc_verbose)
+        std::cout << std::endl << "ERROR:  Belos::LinearProblem failed to set up correctly!" << std::endl;
+      return -1;
+    }
+
+    // *******************************************************************
+    // *************Start the GCRO-DR iteration*************************
+    // *******************************************************************
+
+    // Create an iterative solver manager.
+    RCP< Belos::SolverManager<ST,MV,OP> > newSolver
+      = rcp( new Belos::GCRODRSolMgr<ST,MV,OP>(rcp(&problem,false), rcp(&belosList,false)));
+
+    // **********Print out information about problem*******************
+
+    if (proc_verbose) {
+      std::cout << std::endl << std::endl;
+      std::cout << "Dimension of matrix: " << NumGlobalElements << std::endl;
+      std::cout << "Number of right-hand sides: " << numrhs << std::endl;
+      std::cout << "Max number of restarts allowed: " << maxrestarts << std::endl;
+      std::cout << "Max number of iterations per restart cycle: " << maxiters << std::endl;
+      std::cout << "Relative residual tolerance: " << tol << std::endl;
+      std::cout << std::endl;
+    }
+
+    // Perform solve
+    Belos::ReturnType ret = newSolver->solve();
+
+    // Get the number of iterations for this solve.
+    int numIters = newSolver->getNumIters();
+    std::cout << "Number of iterations performed for this solve: " << numIters << std::endl;
+
+    // Compute actual residuals.
+    bool badRes = false;
+    std::vector<ST> actual_resids( numrhs );
+    std::vector<ST> rhs_norm( numrhs );
+    MV resid(Map, numrhs);
+    OPT::Apply( *A, *X, resid );
+    MVT::MvAddMv( -1.0, resid, 1.0, *B, resid );
+    MVT::MvNorm( resid, actual_resids );
+    MVT::MvNorm( *B, rhs_norm );
+    if (proc_verbose) {
+      std::cout<< "---------- Actual Residuals (normalized) ----------"<<std::endl<<std::endl;
+      for ( int i=0; i<numrhs; i++) {
+        ST actRes = actual_resids[i]/rhs_norm[i];
+        std::cout<<"Problem "<<i<<" : \t"<< actRes <<std::endl;
+        if (actRes > tol) badRes = true;
+      }
+    }
+
+    if (ret!=Belos::Converged || badRes) {
+      success = false;
+      if (proc_verbose)
+        std::cout << std::endl << "ERROR:  Belos did not converge!" << std::endl;
+    } else {
+      success = true;
+      if (proc_verbose)
+        std::cout << std::endl << "SUCCESS:  Belos converged!" << std::endl;
+    }
+  }
+  TEUCHOS_STANDARD_CATCH_STATEMENTS(verbose, std::cerr, success);
+
+  return success ? EXIT_SUCCESS : EXIT_FAILURE;
+  }
+
+int main(int argc, char *argv[]) {
+  // run with different ST
+  run<double>(argc,argv);
+  // run<float>(argc,argv); // FAILS
+  return 0;
```

Comment:
I was thinking about something more like:

```
return run<double>(argc, argv);
```

To get the result of the execution/method `run`

Reply:
Resolved